File: python_backend/game_logic/pathsInfo/get_path.py
from game_logic.step_funcs.step_funcs import *
from game_logic.sqr_case.oob import out_of_bounds
from game_logic.test_objects.sample_board_dicts import sample_board_dicts


def _get_path(board, x, y, take_step):
    """
    travel a path starting at location (x, y) until step out of bounds. add assign_ids
    to list. direction moving depends on which step function take_step is
    """
    x, y = take_step(x, y)
    sqr_list = []
    while not out_of_bounds(x, y):
        sqr_list.append(board[(x, y)])
        x, y = take_step(x, y)

    return sqr_list


if __name__ == "__main__":
    
    # all tests passed

    board = sample_board_dicts['check_example3']

    # test 1, 90 degrees
    print('test 1:')
    sqr_list = _get_path(board, 4, 4, step_1sqr90d)
    print(sqr_list)
    print('\n')

    # test 2, 45 degrees
    print('test 2:')
    sqr_list = _get_path(board, 4, 4, step_1sqr45d)
    print(sqr_list)
    print('\n')

    # test 3, 0 degrees
    print('test 3:')
    sqr_list = _get_path(board, 4, 4, step_1sqr0d)
    print(sqr_list)
    print('\n')

    # test 4, 315 degrees
    print('test 4:')
    sqr_list = _get_path(board, 4, 4, step_1sqr315d)
    print(sqr_list)
    print('\n')

    # test 5, 270 degrees
    print('test 5:')
    sqr_list = _get_path(board, 4, 4, step_1sqr270d)
    print(sqr_list)
    print('\n')

    # test 6, 225 degrees
    print('test 6:')
    sqr_list = _get_path(board, 4, 4, step_1sqr225d)
    print(sqr_list)
    print('\n')

    # test 7, 180 degrees
    print('test 7:')
    sqr_list = _get_path(board, 4, 4, step_1sqr180d)
    print(sqr_list)
    print('\n')

    # test 8, 135 degrees
    print('test 8:')
    sqr_list = _get_path(board, 4, 4, step_1sqr135d)
    print(sqr_list)
    print('\n')

File: app (1)/src/game_logic/pathsInfo/getPath.js
import {outOfBounds} from "../sqrCase/oob";
import {sampleBoardDicts} from "../testObjects/sampleBoardDicts";
import {xyToRf} from "../coordType/crdCnvrt"
import {
    step_1sqr90d,
    step_1sqr45d,
    step_1sqr0d,
    step_1sqr315d,
    step_1sqr270d,
    step_1sqr225d,
    step_1sqr180d,
    step_1sqr135d,
  } from "../helpers/stepFuncs";


export function getPath(board, x, y, takeStep) {
  /**
    travel a path starting at location (x, y) until step out of bounds. add assignIds
    to list. direction moving depends on which step function takeStep is
    */
    var rf, id;
    var sqrList = []; 
    [x, y] = takeStep(x, y);
    while (!outOfBounds(x, y)) {
        rf = xyToRf(x, y);
        id = board[rf];
        sqrList.push(id);
        [x, y] = takeStep(x, y);
    }
    return sqrList;
}

// module.exports = getPath;

/** for node.js
if (require.main === module) {

    // imports 
    import {printBoard} from "../printers/printBoard";
    import {getRfsFromIds} from "../helpers/getRfsFromIds";
    
    let board = sampleBoardDicts["check_example3"];
    var path;

    // test 1, 90 degrees
    sqrList = getPath(board, 4, 4, step_1sqr90d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 1, 90', path, ["d4"])
    console.log("\n");

    // test 2, 45 degrees
    sqrList = getPath(board, 4, 4, step_1sqr45d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 2, 45', path, ["d4"])
    console.log("\n");

    // test 3, 0 degrees
    sqrList = getPath(board, 4, 4, step_1sqr0d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 3, 0', path, ["d4"])
    console.log("\n");

    // test 4, 315 degrees
    sqrList = getPath(board, 4, 4, step_1sqr315d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 4, 315', path, ["d4"])
    console.log("\n");

    // test 5, 270 degrees
    sqrList = getPath(board, 4, 4, step_1sqr270d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 5, 270', path, ["d4"])
    console.log("\n");

    // test 6, 225 degrees
    sqrList = getPath(board, 4, 4, step_1sqr225d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 6, 225', path, ["d4"])
    console.log("\n");

    // test 7, 180 degrees
    sqrList = getPath(board, 4, 4, step_1sqr180d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 7, 180', path, ["d4"])
    console.log("\n");

    // test 8, 135 degrees
    sqrList = getPath(board, 4, 4, step_1sqr135d);
    path = getRfsFromIds(board, sqrList)
    printBoard(board, '\ntest 8, 135', path, ["d4"])
    console.log("\n");

}
*/